src/pages/Trade/Trade.tsx

```diff
@@ -1,18 +1,106 @@
+import { matchPath, Route, Switch, useHistory, useLocation } from 'react-router-dom'
 
 import { Main } from '@/components/Layout/Main'
 import { SEO } from '@/components/Layout/Seo'
+import { LimitOrder } from '@/components/MultiHopTrade/components/LimitOrder/LimitOrder'
+import { LimitOrderRoutePaths } from '@/components/MultiHopTrade/components/LimitOrder/types'
+import { Claim } from '@/components/MultiHopTrade/components/TradeInput/components/Claim/Claim'
+import { ClaimRoutePaths } from '@/components/MultiHopTrade/components/TradeInput/components/Claim/types'
 import { MultiHopTrade } from '@/components/MultiHopTrade/MultiHopTrade'
+import { TradeInputTab, TradeRoutePaths } from '@/components/MultiHopTrade/types'
+import { TRADE_ROUTE_ASSET_SPECIFIC } from '@/Routes/RoutesCommon'
 
 const padding = { base: 0, md: 8 }
 
+type MatchParams = {
+  chainId?: string
+  assetSubId?: string
+  sellAssetSubId?: string
+  sellChainId?: string
+  sellAmountCryptoBaseUnit?: string
+}
+
 export const Trade = memo(() => {
   const translate = useTranslate()
+  const location = useLocation()
+  const tradeInputRef = useRef<HTMLDivElement | null>(null)
+  const methods = useForm({ mode: 'onChange' })
+  const history = useHistory()
+
+  // Extract params directly from location.pathname using matchPath instead of useParams()
+  // Somehow, the route below is overriden by /:chainId/:assetSubId/:nftId, so the wrong pattern matching would be used with useParams()
+  // There is probably a nicer way to make this work by removing assetIdPaths from trade routes in RoutesCommon,
+  // and ensure that other consumers are correctly prefixed with their own route, but spent way too many hours on this and this works for now
+  const match = useMemo(
+    () =>
+      matchPath<MatchParams>(location.pathname, {
+        path: TRADE_ROUTE_ASSET_SPECIFIC,
+        exact: true,
+      }),
+    [location.pathname],
+  )
+
+  const params = match?.params || {}
+
+  const defaultBuyAssetId = useMemo(
+    () =>
+      params.chainId && params.assetSubId ? `${params.chainId}/${params.assetSubId}` : undefined,
+    [params.chainId, params.assetSubId],
+  )
+
+  const defaultSellAssetId = useMemo(
+    () =>
+      params.sellChainId && params.sellAssetSubId
+        ? `${params.sellChainId}/${params.sellAssetSubId}`
+        : undefined,
+    [params.sellChainId, params.sellAssetSubId],
+  )
+
+  const handleChangeTab = useCallback(
+    (newTab: TradeInputTab) => {
+      switch (newTab) {
+        case TradeInputTab.Trade:
+          history.push(TradeRoutePaths.Input)
+          break
+        case TradeInputTab.LimitOrder:
+          history.push(LimitOrderRoutePaths.Input)
+          break
+        case TradeInputTab.Claim:
+          history.push(ClaimRoutePaths.Select)
+          break
+        default:
+          break
+      }
+    },
+    [history],
+  )
+
+  const title = useMemo(() => {
+    if (location.pathname.startsWith(LimitOrderRoutePaths.Input)) {
```

Comment:
Nitpick: this logic would feel more declarative as a switch statement.

Reply:
1eef70ec81